lib/postcode_validation/gateway/pca_address_list.rb

```diff
@@ -0,0 +1,43 @@
+module PostcodeValidation
+  module Gateway
+    class PCAAddressList
+      include HTTParty
+
+      class PCARequestError < PostcodeValidation::Error::RequestError; end
+
+      KEY = ENV['POSTCODE_ANYWHERE_KEY']
+      base_uri 'https://services.postcodeanywhere.co.uk'
```

Comment:
It's not just the strategy that is different unfortunately.

The entire request payload/interface for PCA is different for these two.

```
{
          query: {
            Country: country,
            Key: KEY,
            SearchTerm: search_term,
            Filter: 'None'
          }
        }
```
vs
```
{
          query: {
            Key: KEY,
            Countries: country,
            Text: search_term,
            Filter: 'PostCode'
          }
        }
```

A lot of this complexity stems from PCA itself, as you can see. (Emile)